common/element_handle.go

```diff
@@ -1327,45 +1326,47 @@ func (h *ElementHandle) QueryAll(selector string) []api.ElementHandle {
 	rt := k6common.GetRuntime(h.ctx)
 	parsedSelector, err := NewSelector(selector)
 	if err != nil {
-		k6common.Throw(rt, err)
+		k6Throw(h.ctx, "cannot parse selector %q: %w", selector, err)
 	}
-
 	injected, err := h.execCtx.getInjectedScript(h.ctx)
 	if err != nil {
-		k6common.Throw(rt, err)
-	}
-	pageFn := rt.ToValue(`
-		(injected, selector, scope) => {
-			return injected.querySelectorAll(selector, scope || document, false);
-		}
-	`)
-	opts := evaluateOptions{
-		forceCallable: true,
-		returnByValue: false,
+		k6Throw(h.ctx, "cannot get injected script for selector %q: %w", selector, err)
 	}
 	result, err := h.execCtx.evaluate(
-		h.ctx, opts, pageFn, []goja.Value{
+		h.ctx,
+		evaluateOptions{
+			forceCallable: true,
+			returnByValue: false,
+		},
+		rt.ToValue(`
+			(injected, selector, scope) => {
+				return injected.querySelectorAll(selector, scope || document, false);
+			}
+		`),
+		[]goja.Value{
 			rt.ToValue(injected),
 			rt.ToValue(parsedSelector),
 			rt.ToValue(h),
 		}...)
 	if err != nil {
-		k6common.Throw(rt, err)
+		k6Throw(h.ctx, "cannot evaluate selector %q: %w", selector, err)
 	}
 	if result == nil {
 		return nil
 	}
 
-	arrayHandle := result.(api.JSHandle)
+	var (
+		arrayHandle = result.(api.JSHandle)
+		properties  = arrayHandle.GetProperties()
+		elements    = make([]api.ElementHandle, 0, len(properties))
+	)
 	defer arrayHandle.Dispose()
-	properties := arrayHandle.GetProperties()
-	elements := make([]api.ElementHandle, 0, len(properties))
-	for _, property := range properties {
-		elementHandle := property.AsElement()
-		if elementHandle != nil {
-			result = append(elements, elementHandle)
+	for _, p := range properties {
+		eh := p.AsElement()
+		if eh == nil {
+			p.Dispose()
 		} else {
-			property.Dispose()
+			elements = append(elements, eh)
```

Comment:
This is a single line fix, right? If so, can you split it into a separate commit and all the refactors in another? It's difficult to see what the actual fix is this way.

The two current ones (80818cf and 861cea8) can be squashed.

...

Actually, this fix is already on `main` and was inadvertently part of #196. I noticed it when reviewing, but forgot to comment about it :facepalm:

So I guess you can just fix the merge conflicts here and make this an exclusively refactor PR.

Sidenote: this is why it's risky to mix refactors and fixes, in same commits or in same PRs. They have very different implications and refactors often obscure fixes.

Reply:
I should probably have missed this while separating the other refactoring PR... 🤦 Actually, this is not supposed to be a refactor PR but it has leftover code from the upcoming (not yet exists on Github) unit test PR. There is also a test in this PR for this fix. I'll rebase and see how it goes.

Reply:
Right, then I would make a separate PR for adding the integration test, maybe along with the other unit tests you're working on, and make this one a refactor only.